parametric/gen.py

```diff
@@ -224,22 +224,26 @@ def create_branches(self):
         """Create branches for tree"""
 
         update_log('\nMaking Stems\n')
-
         start_time = time.time()
-        self.branches_curve = bpy.data.curves.new('branches', type='CURVE')
-        self.branches_curve.dimensions = '3D'
-        self.branches_curve.resolution_u = 4
-        self.branches_curve.fill_mode = 'FULL'
-        self.branches_curve.bevel_depth = 1
-        self.branches_curve.bevel_resolution = 10
 
-        # Removed in Blender 2.82+
-        if hasattr(self.branches_curve, 'use_uv_as_generated'):
-            self.branches_curve.use_uv_as_generated = True
+        # Set up container objects and curves for each level
+        for level_name in ['Trunk'] + ['Branches' + str(_) for _ in range(1, self.param.levels)]:
+            level_curve = bpy.data.curves.new(level_name.lower(), type='CURVE')
+            level_curve.dimensions = '3D'
+            level_curve.resolution_u = 4
+            level_curve.fill_mode = 'FULL'
+            level_curve.bevel_depth = 1
+            level_curve.bevel_resolution = 10
```

Comment:
These can certainly be varied per level I think, perhaps for another PR if we are introducing this as a user option?

Reply:
Yeah, that can absolutely be done as a separate PR and is going to be my next one once this is cleaned up.